index.js

```diff
@@ -911,30 +913,36 @@ class Watcher extends EventEmitter {
     if (this.closed) return
     if (this.opened === false) await this._opening
 
-    const snapshot = this.bee.snapshot()
-    this.stream = snapshot.createDiffStream(this.latestDiff, this.range)
+    if (this.current) await this.current.close()
+    if (this.previous) await this.previous.close()
+
+    this.current = this.bee.snapshot()
+    this.previous = this.bee.checkout(this.latestDiff)
```

Comment:
Rather than creating a new checkout, couldn't we use the previous `this.current`? This way it also works with autobase, where the hyperbee can have been truncated in the meantime

Something like

```
await this.previous.close()
this.previous = this.current.snapshot()
await this.current.close()
this.current = this.bee.snapshot()
```

(will be a bit more complex due to the first pass, where `this.current` and `this.previous` don't exist yet)